src/charm.py

```diff
@@ -165,11 +170,26 @@ def __init__(self, *args):
         self.framework.observe(self.on.receive_remote_write_relation_created, self._configure)
         self.framework.observe(self.on.receive_remote_write_relation_changed, self._configure)
         self.framework.observe(self.on.receive_remote_write_relation_broken, self._configure)
+        self.framework.observe(self.on.send_remote_write_relation_broken, self._configure)
+        self.framework.observe(
+            self._remote_write_consumer.on.endpoints_changed,
+            self._on_remote_write_endpoints_changed,
+        )
         self.framework.observe(self.metrics_consumer.on.targets_changed, self._configure)
         self.framework.observe(self.alertmanager_consumer.on.cluster_changed, self._configure)
         self.framework.observe(self.resources_patch.on.patch_failed, self._on_k8s_patch_failed)
         self.framework.observe(self.on.validate_configuration_action, self._on_validate_config)
 
+    def _parse_metric_consumer_alerts(self) -> list:
+        """Parse alert rules from metrics consumer."""
+        alerts = self.metrics_consumer.alerts()
+        consumer_rules_list = []
+        for topology_identifier, alert_rule_groups in alerts.items():
+            consumer_rules_list.extend(alert_rule_groups.get("groups", []))
```

Comment:
This conversion seems like it could be a library concern.
Would it make sense to change the definition of the `extra_alert_rules` so it readily takes the output of `.alerts()`?

Reply:
then we limit it to the functionality of this PR only. While there might be some unforeseeable use cases where use may want just to feed with alerts directly. 
or?

Reply:
I think in general we try to do transactions with standard data structures - the prom alert rule format.
I imagine this would be most future proof. For example, if they add another top-level key to the yaml besides "groups".

Reply:
done, now reload_alerts can consume alerts as from `metrics_consumer`